data/location/src/main/java/com/squirtles/data/location/LocalLocationRepositoryImpl.kt

```diff
@@ -1,18 +1,45 @@
 package com.squirtles.data.location
 
-import android.location.Location
+import android.content.Context
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import com.squirtles.core.model.LocationPoint
 import com.squirtles.domain.location.LocalLocationRepository
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class LocalLocationRepositoryImpl : LocalLocationRepository {
-    private var _currentLocation: MutableStateFlow<Location?> = MutableStateFlow(null)
-    override val lastLocation: StateFlow<Location?> = _currentLocation.asStateFlow()
+class LocalLocationRepositoryImpl @Inject constructor(
+    private val context: Context
+) : LocalLocationRepository {
+    private val Context.dataStore by preferencesDataStore(name = LOCATION_PREFERENCES_NAME)
 
-    override suspend fun saveCurrentLocation(geoLocation: Location) {
-        _currentLocation.emit(geoLocation)
+    private val latKey = stringPreferencesKey(LAT_KEY)
+    private val lngKey = stringPreferencesKey(LNG_KEY)
+
+    override fun readLastLocation(): Flow<LocationPoint?> {
+        return context.dataStore.data.map { preferences ->
+            preferences[latKey]?.let { lat ->
+                preferences[lngKey]?.let { lng ->
+                    LocationPoint(lat.toDouble(), lng.toDouble())
+                }
+            }
```

Comment:
이 부분에서는 중첩된 let을 쓰는 것보다 아래처럼 들여쓰기 없이 명시적으로 쓰는 게 더 잘 읽히는 것 같은데 혹시 어떤가요??

```suggestion
            val lat = preferences[latKey]?.toDoubleOrNull()
            val lng = preferences[lngKey]?.toDoubleOrNull()

            if (lat != null && lng != null) LocationPoint(lat, lng) else null
```

Reply:
좋아요 수정하겠습니다!! 훨씬 읽기 편하고 안전할것같네요